src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -84,6 +84,31 @@ public boolean isSameAppointment(Appointment otherAppointment) {
                 && otherAppointment.timePeriod.equals(timePeriod);
     }
 
+    /**
+     * Returns true if both appointments has overlapping time period.
+     * Overlapping is defined as within, the same or partly coinciding timePeriods.
+     */
+    public boolean hasOverlappingTimePeriod(Appointment otherAppt) {
+        Time existingStartTime = this.getStartTime();
+        Time existingEndTime = this.getEndTime();
+        Time otherStartTime = otherAppt.getStartTime();
+        Time otherEndTime = otherAppt.getEndTime();
+
+        // other startTime is before existing startTime &&
+        // other endTime is before or equal to existing startTime --> return false
+        if (otherStartTime.compareTo(existingStartTime) < 0 && otherEndTime.compareTo(existingStartTime) <= 0) {
```

Comment:
Just was wondering if we need the first check for otherStartTime since otherEndTime > otherStartTime. But I think it's fine to leave it in as well. No harm with extra guards.

Reply:
I'll remove it and do the check in TimePeriod. Thanks!